### PR TITLE
Feature/security analysis test cases creator

### PR DESCRIPTION
**Pull request by tlec-artelys**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
- Added internal classes to provide contingencies and state monitors

**What is the current behavior?** *(You can also link to an open issue here)*
Files containing contingencies and state monitors are inputs to the creation of security analysis test cases.

**What is the new behavior (if this is a feature change)?**
Creates test cases for security analysis without specifying files for the contingencies or the state monitors.
The creation of new test cases generates the contingencies and state monitors files.

### CODE REVIEW

File: src/main/java/com/powsybl/integrationtest/creation/security/statemonitors/DefaultStateMonitorsProvider.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            }
+
+            // Add sample of this type of element to the sample
+            sample.put(clazz, elementsSample);
+        });
+
+        return sample;
+    }
+
+    public List<StateMonitor> createStateMonitorList(Network network) {
```
Comment (armic-artelys) [inline src/main/java/com/powsybl/integrationtest/creation/security/statemonitors/DefaultStateMonitorsProvider.java:73]: Essaie de ne pas oublier les @Override (pas absolument nécessaires mais aide à la lecture)

File: src/main/java/com/powsybl/integrationtest/creation/security/statemonitors/DefaultStateMonitorsProvider.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        // Get network's elements that can be monitored (branches, voltage levels and three windings transformers)
+        Iterable<VoltageLevel> voltageLevels = network.getVoltageLevels();
+        Iterable<ThreeWindingsTransformer> threeWindingsTransformers = network.getThreeWindingsTransformers();
+        Iterable<Branch> branches = network.getBranches();
```
Comment (armic-artelys) [inline src/main/java/com/powsybl/integrationtest/creation/security/statemonitors/DefaultStateMonitorsProvider.java:77]: Ici aussi tu peux faire le tout en une seule ligne et réduire la taille générale de la méthode.
Ex:
```java
network.getVoltageLevels().forEach(vl -> voltageLevelIds.add(vl.getId()));
```

**Review by armic-artelys — CHANGES_REQUESTED** [review]
Il faut modifier des choses (voir mes commentaires)

**Review by armic-artelys — COMMENTED** [review]
OK pour moi (à part un petit détail pas très important)
@obrix, je te laisse nous dire ce que tu en penses.

**Review by armic-artelys — CHANGES_REQUESTED** [review]
Toujours des problèmes dans la classe DefaultContingenciesProvider

File: src/main/java/com/powsybl/integrationtest/creation/security/contingencies/DefaultContingenciesProvider.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.integrationtest.creation.security.contingencies;
+
+import com.powsybl.contingency.Contingency;
+import com.powsybl.contingency.ContingencyElement;
+import com.powsybl.contingency.LineContingency;
+import com.powsybl.iidm.network.*;
+import org.jgrapht.alg.util.Pair;
```
[the remark this answers is not in the document]
Comment (obrix) [reply src/main/java/com/powsybl/integrationtest/creation/security/contingencies/DefaultContingenciesProvider.java:13]: If Pairs are really necessary the implementations from org.apache.commons.math3.util and org.apache.commons.lang3.tuple are used within powsybl-core. I think we should stick with the one of them (the on from lang package seems more used.

File: src/main/java/com/powsybl/integrationtest/creation/security/contingencies/ContingenciesProvider.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import com.powsybl.iidm.network.Network;
+
+import java.util.List;
+
+/**
+ * Component used to create security analysis test cases, as it provides a list of contingencies.
+ *
+ * @author Théo Le Colleter <theo.le-colleter at artelys.com>
+ */
+public interface ContingenciesProvider {
```
Comment (obrix) [inline src/main/java/com/powsybl/integrationtest/creation/security/contingencies/ContingenciesProvider.java:19]: I think you can reuse ContingenciesProvider from powsybl-core that has the exact same prototype. If you absolutely want to use a different interface maybe name it differently.

Comment (tlec-artelys) [reply src/main/java/com/powsybl/integrationtest/creation/security/contingencies/ContingenciesProvider.java]: I removed the interface I had created, and I reuse the one in contingency-api.
Thank you.

File: src/main/java/com/powsybl/integrationtest/creation/security/SATestcaseCreator.java [hunk 5]
```diff
@@ -74,29 +75,18 @@ public void createResults(String exportName, Network network, SecurityAnalysisPa
...
-
-    private void createStateMonitors(String exportName, Network network, Path outputDir) throws IOException {
-        final List<StateMonitor> stateMonitors = stateMonitorsProvider.createStateMonitorList(network);
-        this.stateMonitors = stateMonitors;
 
         // Export state monitors in a .json file
-        ObjectMapper mapper = JsonUtil.createObjectMapper();
-        OutputStream outputStream = Files.newOutputStream(outputDir.resolve(exportName + ".json"));
-        mapper.writeValue(outputStream, stateMonitors);
+        ObjectMapper mapperStateMonitors = JsonUtil.createObjectMapper();
```
[the remark this answers is not in the document]
Comment (tlec-artelys) [reply src/main/java/com/powsybl/integrationtest/creation/security/SATestcaseCreator.java]: Si tout à fait, merci.

File: src/main/java/com/powsybl/integrationtest/creation/security/contingencies/DefaultContingenciesProvider.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (armic-artelys) [reply src/main/java/com/powsybl/integrationtest/creation/security/contingencies/DefaultContingenciesProvider.java]: I'm not against the use of Pairs, but rather against the use of a class that comes from a graph-related lib, while doing absolutely nothing graph-related here. We should not have an explicit dependency to jgrapht.

**Review by armic-artelys — APPROVED** [review]
OK pour moi

**Pull request merged**